### DISCUSSION TITLE
Is this gonna work with Api ?

### DISCUSSION THREAD

**Discussion by Danger-Mkh** [qruto/laravel-wave#71] [Q&A]
For my use case i needed to use this package for API.

The public channel works as expected, but private and presence channels don't.

I think it's because `/wave` endpoint doesn't aware of authenticated user using Authorization header.

So i handle it with a `Middlerware`, check if authorization header exists, then set the `Auth` guard to sanctum and everything work as expected.

**Comment by slavarazum** [comment]
Hi @Danger-Mkh !

Do you use Echo on the client side, or how do you send /wave requests?

It should be sent with appropriate auth headers.

**Reply by Danger-Mkh** [reply]
Hey @slavarazum,

Yes, I use Echo on the client side.
This is exactly what I'm using:
```js
new Echo({
    broadcaster: WaveConnector,
    authEndpoint: BASE_URL + "/api/broadcasting/auth",
    endpoint: BASE_URL + "/wave",
    request: {
        headers: {
            Accept: 'application/json',
            Authorization: 'Bearer ' + token,
        }
    },
    bearerToken: token,
});
```

Also I added that `/api/broadcasting/auth` endpoint.
I put this inside `api.php`

```php
Broadcast::routes(['middleware' => ['auth:sanctum']]);
```

**Reply by Danger-Mkh** [reply]
And this is the Middleware I write and added to `wave` config file.
```php
<?php

namespace App\Http\Middleware;

use Closure;
use Illuminate\Http\Request;
use Symfony\Component\HttpFoundation\Response;

class FindUserByAuthorizationHeaderIfExists
{
    /**
     * Handle an incoming request.
     *
     * @param  \Closure(\Illuminate\Http\Request): (\Symfony\Component\HttpFoundation\Response)  $next
     */
    public function handle(Request $request, Closure $next): Response
    {
        $beare_token = $request->bearerToken();
        if (filled($beare_token)) {
            auth()->shouldUse('sanctum');
        }

        return $next($request);
    }
}

```

**Reply by slavarazum** [reply]
I'm not sure for what purpose do you use different auth endpoint and Bearer token at the same time.

Echo with Wave broadcaster handling authentication by default. If you want to bring your own authentication endpoint, you have to write it by your own.

P.S. looks like you are only checking if the Bearer token is filled 👀

**Reply by Danger-Mkh** [reply]
The `/wave` endpoint couldn't find the authenticated user.
Because it used session driver authentication.

For my use case I wanted to use `/wave` endpoint as authenticated user and guest user at same time.


> P.S. looks like you are only checking if the Bearer token is filled 👀

The rest of authentication are on Laravel, just need to inform to use `sanctum` driver.
